pkg/apis/serving/v1alpha2/inferenceservice_types.go

```diff
@@ -171,6 +171,8 @@ type TensorRTSpec struct {
 type XGBoostSpec struct {
 	// The location of the trained model
 	StorageURI string `json:"storageUri"`
+	// Number of thread to be used by XGBoost
+	NThread int `json:"nthread,omitempty"`
```

Comment:
should we default this with CPU resource ?

Reply:
+1

Reply:
do you mean not to expose it as configuration to user? I am just worried that setting it to CPU resources will not always be the best for performance. 
I did a benchmark on inference service with CPU request of 2 comparing nthread=1 & nthread=2, the result was that nthread=1 is outperforming nthread=2 


nthread=1

QPS | mean | 50 | 90 | 95 | 99 | max
-- | -- | -- | -- | -- | -- | --
10 | 6.392413ms | 6.276356ms | 7.534008ms | 7.893341ms | 9.63982ms | 27.36586ms
50 | 5.579441ms | 5.507205ms | 6.549779ms | 6.912419ms | 7.934989ms | 50.055502ms
100 | 5.361426ms | 5.163966ms | 6.146815ms | 6.629187ms | 8.5592ms | 75.016721ms
500 | 4.300006ms | 3.306792ms | 4.934485ms | 6.137519ms | 35.665023ms | 126.992667ms


nthread=2 (same as CPU request)

QPS | mean | 50 | 90 | 95 | 99 | max
-- | -- | -- | -- | -- | -- | --
10 | 7.113759ms | 7.032882ms | 8.017496ms | 8.330207ms | 10.350565ms | 34.635795ms
50 | 7.419224ms | 6.735612ms | 8.019301ms | 8.952701ms | 36.417826ms | 69.477059ms
100 | 7.044067ms | 5.812522ms | 7.246207ms | 9.12106ms | 49.487041ms | 82.183296ms
500 | 18.847475ms | 6.938928ms | 49.667048ms | 72.179678ms | 124.653565ms | 288.256637ms

Reply:
We clearly need to do some fancier defaults here. Can you cut an issue?

Reply:
https://github.com/dmlc/xgboost/issues/3042 FYI

Reply:
I created an issue here https://github.com/kubeflow/kfserving/issues/584